gmp library linking error in Makefile

On line 8 of Makefile, cc should take -lgmp option later than source code.

Otherwise undefined reference error can occur.
```
/tmp/ccDyHO5q.o: In function `field_init':
ssss.c:(.text+0x1a8): undefined reference to `__gmpz_init_set_ui'
ssss.c:(.text+0x1b5): undefined reference to `__gmpz_setbit'
...
```
[Related note](http://stackoverflow.com/a/11894098)

--- a/Makefile
+++ b/Makefile
@@ -5,7 +5,7 @@ compile: ssss-split ssss-combine
 doc: ssss.1 ssss.1.html
 
 ssss-split: ssss.c
-	$(CC) -W -Wall -O2 -lgmp -o ssss-split ssss.c
+	$(CC) -W -Wall -O2 -o ssss-split ssss.c -lgmp
 	strip ssss-split
 
 ssss-combine: ssss-split

